### ISSUE TITLE
Failure to parse regex correctly

### ISSUE THREAD

**Issue by BoredBored** [hanickadot/compile-time-regular-expressions#174]
CTRE thinks the following regex has a syntax error:
```
 R"([rRfFUu]{0,2}?("{3}|")((?:[^\\"]|\\.|\\)*\1)?|)"    //capture " strings
R"(([rRfFUu]{0,2}?('{3}|')((?:[^\\']|\\.|\\)*\4)?)|)"   //capture ' strings
R"((#[^\r\n]*)|)"                                       //capture comments
R"(([\n\r][ \t]*)|)"                                    //capture newlines
R"((\\[^\r\n]*)|)"                                      /* capture \TheBackslashAndAnythingAfterIt  */
R"(()"
    R"((\.{3})|)"                                                  //capture "..."
    R"((->)|)"                                                     //capture ->
    R"(([0-9][0-9_]*\.[0-9_]*[0-9][0-9_]*[a-zA-Z]*[0-9_]*)|)"       //──┬┬─> floating point numbers
    R"(([0-9][0-9_]*\.[a-zA-Z]*[0-9_]*)|)"                          //──┘│
    R"((\.[0-9][0-9_]*[a-zA-Z]*[0-9_]*)|)"                          //───┘
    R"(([<>*\/]{2})=?|)"                                    //capture 2-3 character operators
    R"(([!%&*+\-<=>@\/\\^|:]=))"                            //capture 2 caracter operators
R"()|)"
R"(([!-\/:-@\[-^{-~]|[^\s!-\/:-@\[-^{-~]+))"            //capture anything else
```
https://regex101.com/r/Vn2DFE/1/

The above regex tokenizes the python language
from https://github.com/BoredBored/Turtle/blob/6b7692fb7f9d637ac66ad5cebd14e652ccdbd782/token.hpp#L101

**Comment by hanickadot** [comment]
small fix for pipe in character class pushed into `main` as d1aa6a6bdfcd96d9732e393764ac7e309c1bd565

**Comment by hanickadot** [comment]
second part of fix in 0fccc73debc48dc7292f888561efec2896a3827f

**Comment by hanickadot** [comment]
fixed
